packages/ckeditor5-engine/src/conversion/mapper.ts

```diff
@@ -1155,53 +1150,54 @@ export class MapperCache extends /* #__PURE__ */ EmitterMixin() {
 	}
 
 	/**
-	 * Clears all the stored cache starting before given `viewNode`. The `viewNode` can be any node that is inside a tracked view element
+	 * Clears all the stored cache that is after given `viewNode`. The `viewNode` can be any node that is inside a tracked view element
 	 * or view document fragment.
+	 *
+	 * In reality, this function may clear a bit more cache than just "starting after" `viewNode`, but it is guaranteed that at least
+	 * all cache after `viewNode` is invalidated.
 	 */
-	private _clearCacheStartingBefore( viewNode: ViewNode ): void {
+	private _clearCacheAfter( viewNode: ViewNode ): void {
 		// To quickly invalidate the cache, we base on the cache list index stored with the node. See docs for `this._nodeToCacheListIndex`.
 		const cacheListIndex = this._nodeToCacheListIndex.get( viewNode );
 
 		// If there is no index stored, it means that this `viewNode` has not been cached yet.
 		if ( cacheListIndex === undefined ) {
-			// If the node is not cached, maybe it's parent is. We will try to invalidate the cache starting from before the parent.
-			// Note, that there always must be a parent if we got here.
+			// If the node is not cached, maybe it's parent is. We will try to invalidate the cache using the parent.
 			const viewParent = viewNode.parent as ViewElement;
 
-			// If the parent is a non-tracked element, try clearing the cache starting before it.
+			// If the parent is a non-tracked element, try clearing the cache starting from the position before it.
 			//
-			// This situation may happen e.g. if structure like `<p><strong><em>Foo</em></strong>...` was stepped over in
-			// `Mapper#_findPositionIn()` and the children are not cached yet, but the `<strong>` element is. If something changes
-			// inside this structure, make sure to invalidate all the cache after `<strong>`.
+			// For example: `<p>Abc<strong>def<em>ghi</em></strong></p>`.
+			//
+			// If `viewNode` is `<em>` in this case, and it was not cached yet, we will try to clear cache starting from before `<strong>`.
 			//
 			// If the parent is a tracked element, then it means there's no cache to clear (nothing after the element is cached).
-			// In this case, there's nothing to do.
+			// In this case, there's nothing to do. We assume that there are no "holes" in caching in direct children of tracked element
+			// (that is if some children is cached, then its previous sibling is cached too, and we would not end up inside this `if`).
+			//
+			// TODO: Most probably this `if` could be removed altogether, after recent changes in Mapper.
+			// TODO: Now we cache all items one after another, so there aren't any "holes" anywhere, not only on top-level.
 			//
 			if ( !this._cachedMapping.has( viewParent ) ) {
-				this._clearCacheStartingBefore( viewParent );
+				this._clearCacheInsideParent( viewParent.parent!, viewParent.index! );
```

Comment:
This is another place where the same fix was introduced, although given the latest changes in `Mapper` it seems that this code may not be reachable outside of artificial unit tests crafted for `MapperCache`. As in "TODO" above, it may be a dead code.